contracts/Vault.sol

```diff
@@ -46,6 +46,7 @@ contract Vault is
     //
 
     bytes32 public constant INVESTOR_ROLE = keccak256("INVESTOR_ROLE");
+    bytes32 public constant HARVESTOR_ROLE = keccak256("HARVESTOR_ROLE");
```

Comment:
What would happen if we let anyone call the harvest function?

Reply:
the reason we decided to harvest fee is that we keep fee in contract, not immediately send to treasury.
if there is no permission, anybody can claim fee to reduce tvl.

Reply:
I see. It makes sense; you can keep the money around 👍  about the role, why not use the INVESTOR_ROLE? that's already used to control when the money is invested

Reply:
we could use INVESTOR_ROLE, however to make sure roles, i added another role.

Reply:
I don't understand. the fee is always sent to `treasury`. it doesn't matter who calls the function